ethereum/core/src/main/java/org/hyperledger/besu/ethereum/trie/diffbased/verkle/VerkleAccount.java

```diff
@@ -94,48 +94,58 @@ public static VerkleAccount fromRLP(
     final RLPInput in = RLP.input(encoded);
     in.enterList();
 
-    final long nonce = in.readLongScalar();
     final Wei balance = Wei.of(in.readUInt256Scalar());
-    final Hash storageRoot = Hash.wrap(in.readBytes32());
+    final long nonce = in.readLongScalar();
     final Hash codeHash = Hash.wrap(in.readBytes32());
-
+    final long codeSize = in.readLongScalar();
     in.leaveList();
 
     return new VerkleAccount(
-        context, address, address.addressHash(), nonce, balance, storageRoot, codeHash, mutable);
+        context, address, address.addressHash(), nonce, balance, codeSize, codeHash, mutable);
+  }
+
+  @Override
+  public Optional<Long> getCodeSize() {
+    return Optional.of(codeSize);
+  }
+
+  @Override
+  public void setCode(final Bytes code) {
+    super.setCode(code);
+    if (code == null || code.isEmpty()) {
+      this.codeSize = 0L;
+    } else {
+      this.codeSize = code.size();
+    }
+  }
```

Comment:
So there is a kind of code smell to me here. When you construct `VerkleAccount` you have a `codeSize != 0` but `getCode() == null` this is not clear at all unless you are totally familiar with Verkle. And even then...

I understand that `code` and `codeSize` are 2 different regions of the DB and so is `codeHash` and that performance wise we should avoid touching slots unnecessarily but why not having a code provider with a lambda function instead of the field `Code code` that only populates it and goes to the DB if someone calls `getCode`? That would be more intuitive to me and consistent. Don't you know at least a way to get the code from the DB when constructing a `VerkleAccount` ?

Reply:
It's already working like that, it's using the same code as Bonsai, VerkleAccount is calling the context (MutableWorldstate) to retrieve the code when we call getCode 

See in DiffBasedAccount
```java
 public Bytes getCode() {
    if (code == null) {
      code = context.getCode(address, codeHash).orElse(Bytes.EMPTY);
    }
    return code;
  }
```

Reply:
Yes I understand, this is not something just with Verkle but Bonsai also has the same problem. Maybe nothing to do for now, but it would be better to clarify.

Right now, there's so many ways you can set code in both: using a `context`, calling `setCode` directly, doing a copy constructor way. I feel this has potential to create problems and at least it's hard for me to deterministically say how the code is set into an account. Why can't you have code set always at construct time? Agree needs some thought though.

But maybe it's because we're saving on object allocation and reusing `*Account` objects?

Reply:
it's an optmization, we don't have to read a code if we don't need it . the idea is to read in the database only when we really want to access the code

also I don't see a potential problem with that. The context is used when you read the code from the database, after when you modify the code you will always pass via the account. the worldstate (via accumulator) will check the code in the account and save in the database

Reply:
I think I will have in the futur to explain all of this logic in a doc (accumulator, worldstate, updater etc)